src/proxy.rs

```diff
@@ -12,6 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::collections::HashSet;
```

Comment:
Is hash necessary? Could we just use a vec, sort it once during init and then efficiently search it without calculating hashes on the u16. Because its small we could probably just iterate for a match instead of sort/binary search even

Reply:
Probably not, but its just 1 instance with like 4 keys and we Arc it anyways so doesn't seem so bad?

Reply:
Yeah, if I start feeling froggy maybe I'll test it and let ya know.